src/unix.rs

```diff
@@ -42,6 +45,17 @@ fn dup_fd(fd: RawFd) -> io::Result<Stdio> {
     dup_result.map(File::into_stdio)
 }
 
+fn nix_err_to_io_err(err: nix::Error) -> io::Error {
+    match err {
+        Sys(err_no) => {
+            io::Error::from(err_no)
+        }
+        _ => {
+            io::Error::new(ErrorKind::InvalidData, err)
```

Comment:
I think I would rather this be a `panic!`, since it's really not expected, and if it ever starts happening we want to know about it.